projects/com.oracle.truffle.llvm.types/src/com/oracle/truffle/llvm/types/memory/LLVMStack.java

```diff
@@ -76,6 +76,10 @@ public void free() {
         isFreed = true;
     }
 
+    public static long allocate(long size) {
```

Comment:
For integration of this PR we need to have an own stack for signal handlers. I guess we also have to lock on it once we enter a signal handler to avoid race conditions.